File: pruebasProyecto/jet.cpp
#include "jet.h"

// Implementación del constructor
PrivateJet::PrivateJet(const string& _brand, const string& _model, const string& _id,  int _capacity, int _maxSpeed,
             int _autonomy, int _year, const string& _condition, const string& _ubication,
             bool _availability,  Person *_owner)
    : Aircraft(_brand, _model, _id, _capacity, _maxSpeed, _autonomy, _year, _condition, _ubication, _availability) {
    owner = _owner;
}

PrivateJet::PrivateJet(){}

// Implementación de métodos set
void PrivateJet::setOwner( Person *_owner) {
    owner = _owner;
}

void PrivateJet::addService(const string& service) {
    services.push_back(service);
}

void PrivateJet::addDestination(const string& destination) {
    destinations.push_back(destination);
}

// Implementación de métodos get
Person* PrivateJet::getOwner() {
    return owner;
}

vector<string> PrivateJet::getServices() {
    return services;
}

vector<string> PrivateJet::getDestinations() {
    return destinations;
}

void Jet::print() const {
    cout << "Jet Information:" << endl;
    Aircraft::print();
    if (owner != nullptr) {
        cout << "Owner Information:" << endl;
        owner->print();
    }
    

    cout << "Services Offered:" << endl;
    int serviceIndex = 1;
    for (int i = 0; i < services.size(); ++i) {
        cout << serviceIndex << ". " << services[i] << endl;
        ++serviceIndex;
    }

    cout << "Destinations Offered:" << endl;
    int destinationIndex = 1;
    for (int i = 0; i < destinations.size(); ++i) {
        cout << destinationIndex << ". " << destinations[i] << endl;
        ++destinationIndex;
    }
}

File: proyectoPython/jet.py
from aircraft import Aircraft

class PrivateJet(Aircraft):
    def __init__(self,  brand= None,  model= None,  ident= None,  capacity= None,  max_speed= None,  autonomy= None,  year= None,  condition= None,  ubication= None,  owner = None, _services = None, _destinations = None):
        super().__init__(brand, model, ident, capacity, max_speed, autonomy, year, condition, ubication)
        self.owner = owner
        self.services = _services
        self.destinations = _destinations

    def set_owner(self, owner):
        self.owner = owner

    def add_service(self, service):
        self.services.append(service)

    def add_destination(self, destination):
        self.destinations.append(destination)

    def get_owner(self):
        return self.owner

    def get_services(self):
        return self.services

    def get_destinations(self):
        return self.destinations